src/main/java/de/tum/in/www1/artemis/repository/ExamRepository.java

```diff
@@ -53,11 +53,19 @@
     Optional<Exam> findWithStudentExamsExercisesById(Long id);
 
     @Query("""
-            select e
-            from Exam e, User u
-            where u.id = :#{#userId} and e.course.instructorGroupName member of u.groups
+            select distinct e
+            from Exam e left join fetch e.exerciseGroups eg left join fetch eg.exercises ex, User u
+            where u.id = :#{#userId} and e.course.instructorGroupName member of u.groups and
+            TYPE(ex) = QuizExercise
             """)
-    List<Exam> getExamsForWhichUserHasInstructorAccess(@Param("userId") Long userId);
+    List<Exam> getExamsWithQuizExercisesForWhichUserHasInstructorAccess(@Param("userId") Long userId);
```

Comment:
We can avoid fetching the entire user table here by providing the user groups directly as parameters.
Example:
https://github.com/ls1intum/Artemis/pull/2706/files#diff-c8981cc5e7f04e7084e297989da9a545bdc3d79c630b484bdcfcbfafaf26bea2R286
https://github.com/ls1intum/Artemis/pull/2706/files#diff-0309f85b5d39e342c0bf565948a617fd245016223c9b2aa70db88eef0790df39R113

Reply:
Good catch, Lukas! Adapted in 3551685